### PR TITLE
*: ensure that log and error messages start with lowercase

### PR DESCRIPTION
**Pull request by knz**
This commit adjusts a few log and error calls to ensure the texts
start with a lowercase letter. (Messages that start with a code
identifier keep their capital if any - these are treated as proper
nouns.)

Good: `log.Infof(ctx, "this is my message: %s", err)`
Less good: `log.Info(ctx, "This is my message: %s", err)`

Good: `errors.Wrap(err, "some context")`
Less good: `errors.Wrap(err, "Some context")`

This aligns the code with the prevalent convention valid throughout
the project. The reason for adjusting this now is that log entries
will soon receive additional scrutiny from users in the shape of
documentation. We want the UX to be consistent throughout all logging
and error outputs.

A linter may be introduced later to enforce this convention.

For context, why we use lowercase:

Both log and error messages are always reported to users in a text
format after a colon, for example, `file.go:123: this is the log
message`. 
In English, random capital letters are not allowed in the
middle of sentences (unlike in German) and distract from the flow of
reading.

This is particularly important for error objects, which can receive an
arbitrary amount of text prefixes using `errors.Wrap()` in
callers. Additionally, error objects can be included inside a larger
sentence, for example `while handling "this is my error"`. Here, as
well, capitals are not welcome after an opening quote in the middle of
a sentence.

Release note: None

### CODE REVIEW

File: pkg/jobs/jobs.go [hunk 1]
```diff
@@ -329,7 +329,7 @@ func (j *Job) FractionProgressed(ctx context.Context, progressedFn FractionProgr
 		}
 		if fractionCompleted < 0.0 || fractionCompleted > 1.0 {
 			return errors.Errorf(
-				"Job: fractionCompleted %f is outside allowable range [0.0, 1.0] (job %d)",
+				"job: fractionCompleted %f is outside allowable range [0.0, 1.0] (job %d)",
```
Comment (spaskob) [inline pkg/jobs/jobs.go:332]: please change to
`job %d: fractionCompleted %f is outside allowable range [0.0, 1.0]`

File: pkg/jobs/jobs.go [hunk 2]
```diff
@@ -355,7 +355,7 @@ func (j *Job) HighWaterProgressed(ctx context.Context, progressedFn HighWaterPro
 		}
 		if highWater.Less(hlc.Timestamp{}) {
 			return errors.Errorf(
-				"Job: high-water %s is outside allowable range > 0.0 (job %d)",
+				"job: high-water %s is outside allowable range > 0.0 (job %d)",
```
Comment (spaskob) [inline pkg/jobs/jobs.go:358]: Same here:
`job %d: high-water %s is outside allowable range > 0.0`

File: pkg/jobs/update.go [hunk 3]
```diff
@@ -118,7 +118,7 @@ func (j *Job) Update(ctx context.Context, updateFn UpdateFn) error {
 
 		statusString, ok := row[0].(*tree.DString)
 		if !ok {
-			return errors.Errorf("Job: expected string status on job %d, but got %T", *j.id, statusString)
+			return errors.Errorf("job: expected string status on job %d, but got %T", *j.id, statusString)
```
Comment (spaskob) [inline pkg/jobs/update.go:121]: similarly here

File: pkg/jobs/update.go [hunk 4]
```diff
@@ -194,7 +194,7 @@ func (j *Job) Update(ctx context.Context, updateFn UpdateFn) error {
 		}
 		if n != 1 {
 			return errors.Errorf(
-				"Job: expected exactly one row affected, but %d rows affected by job update", n,
+				"job: expected exactly one row affected, but %d rows affected by job update", n,
```
Comment (spaskob) [inline pkg/jobs/update.go:197]: and here

**Review by spaskob — APPROVED** [review]

**Review by irfansharif — APPROVED** [review]

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52718)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @spaskob)

___
*[pkg/jobs/jobs.go, line 332 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52718#-MEYzm1-LSwaAtFNT0sp-r1-332:-MFGUGo17a__8eh8ouab:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/009fcf565674574df3752a5426567c8b6ba5d0d5/pkg/jobs/jobs.go#L332)):*
<details><summary><i>Previously, spaskob (Spas Bojanov) wrote…</i></summary><blockquote>

please change to
`job %d: fractionCompleted %f is outside allowable range [0.0, 1.0]`
</blockquote></details>

Done.
___
*[pkg/jobs/jobs.go, line 358 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52718#-MEYzm1-LSwaAtFNT0sp-r1-358:-MFGUK9u3Cv1SzFxcLO4:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/009fcf565674574df3752a5426567c8b6ba5d0d5/pkg/jobs/jobs.go#L358)):*
<details><summary><i>Previously, spaskob (Spas Bojanov) wrote…</i></summary><blockquote>

Same here:
`job %d: high-water %s is outside allowable range > 0.0`
</blockquote></details>

Done.
___
*[pkg/jobs/update.go, line 121 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52718#-MEYzm1-LSwaAtFNT0sr-r1-121:-MFGUVhI-hGJ8Aea8r-K:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/009fcf565674574df3752a5426567c8b6ba5d0d5/pkg/jobs/update.go#L121)):*
<details><summary><i>Previously, spaskob (Spas Bojanov) wrote…</i></summary><blockquote>

similarly here
</blockquote></details>

Done.
___
*[pkg/jobs/update.go, line 197 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52718#-MEYzm1-LSwaAtFNT0sr-r1-197:-MFGUc335E7xZZuFE0pN:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/009fcf565674574df3752a5426567c8b6ba5d0d5/pkg/jobs/update.go#L197)):*
<details><summary><i>Previously, spaskob (Spas Bojanov) wrote…</i></summary><blockquote>

and here
</blockquote></details>

Done.

**Pull request merged**